src/scheduler/serialization.rs

```diff
@@ -128,12 +131,23 @@ pub(crate) fn serialize_schedule(schedule: &Schedule) -> String {
     buf.write_u64_varint(schedule.len() as u64).unwrap();
     buf.write_u64_varint(schedule.seed).unwrap();
     buf.extend(encoded.as_raw_slice());
-    hex::encode(buf)
+
+    let num_line_feeds = (buf.len() + BYTES_PER_LINE - 1) / BYTES_PER_LINE;
+    let mut hex_buf = vec![0; buf.len() * 2 + num_line_feeds];
+    let mut offset = 0;
+    for chunk in buf.chunks(BYTES_PER_LINE) {
+        hex::encode_to_slice(chunk, &mut hex_buf[offset..offset + chunk.len() * 2]).unwrap();
+        offset += chunk.len() * 2;
+        hex_buf[offset] = LINE_FEED;
+        offset += 1;
+    }
+    String::from_utf8(hex_buf).unwrap()
```

Comment:
This is a little worse allocation-wise, but probably clearer, and doesn't add a new line at the end:
```suggestion
    let serialized = hex::encode(buf);
    let lines = serialized.as_bytes().chunks(BYTES_PER_LINE).collect::<Vec<_>>();
    let wrapped = lines.join(&['\n' as u8][..]);
    String::from_utf8(wrapped).unwrap()
```

Reply:
Haha, I contemplated descending to unsafe to skip zeroing the buffer and using `from_utf8_unchecked`. But fair enough, this is not performance critical, so accepting the extra allocations should be fine. Same as the extra allocation we now have in `deserialize_schedule` to remove newlines.